src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -151,6 +151,11 @@ public static unsafe ulong BigMul(ulong a, ulong b, out ulong low)
                 low = tmp;
                 return high;
             }
+            else if (ArmBase.Arm64.IsSupported)
+            {
+                low = a * b;
```

Comment:
This is probably a code pattern that the JIT should handle and generate `UMULL` and `SMULL`

This has been previously blocked due to codegen issues, but I think the work Carol did around multi-reg returns unblocked that, in which case these should forward to an intrinsic `(long, long) BigMul(long, long)`

Reply:
@tannergooding I am not sure I understand you comment about handling and generating `umull`/`smull` in context of `BigMul(long, long)`

Reply:
My mistake, I misread and thought `UMULL` had a variant that multiplies two 64-bit and returns the 128-bit result in 2x register like `IMUL` and `MUL` support on x86/x64.

It looks like it's just `32x32=64` and `64x64=upper 64` (the latter via `UMULH`). I don't see a variant that also returns the `lower 64` with the same computation.

Reply:
It actually looks like you might be able to achieve it using `PMULL` and/or `PMULL2`, but it isn't clear if that's "always a win"

Reply:
I don't believe `PMULL{2}` can be used at this context - results of the instructions (polynomial multiplication) does't correspond to the result of integer multiplications.

But there is a definitely room for improving 32-bit integer multiplication (see #47490) that would result in more efficient code for `BigMul(int, int)`